### DISCUSSION TITLE
Kodi and subtitles

### DISCUSSION THREAD

**Discussion by Guguloi** [moneytoo/Player#202] [Ideas]
Just found your player man and I can say I'm impressed. Integrated with Kodi on my 55XF9005 and it's the only one which can play DV (mp4 and mkv) and extra large 4k files (100GB+) without problems so kudos for this. 

The only drawback for me are subtitles. Instead opening a location for subtitles if you can use (somehow) Kodi subtitles addons it will be perfect.
Kodi player just opens my subtitles addon and I select a subtitle and that's it. No download, no hassle.
For Just Player I must download subtitle, transfer it to TV, opened it and deleted it after movie/show watched. A little bit complicated.
Don't know  if it's possible but it will be nice to have this.

Anyway, thanks for your work you have an amazing player here.

**Comment by moneytoo** [comment]
If you could make the subtitle addon actually download and save it next to the video file, Just Player should be able to pick it up and load automatically.

**Comment by Guguloi** [comment]
How can I do that?
Those are my subtitles settings in KODI player.
![IMG_20211107_190451](https://user-images.githubusercontent.com/89657324/140654731-79850df3-514c-4c46-9ae7-29991a8d797b.jpg)
And if I chose subtitle icon during playing movie, then Download subtitle I have all those subtitles services to choose from.
![IMG_20211107_190820](https://user-images.githubusercontent.com/89657324/140654787-2204b79a-d87d-4e15-83e1-66e21149ce39.jpg)
I want something similar in Just Player. Opening subtitles icon then show all those subtitles services from KODI.
But I have a feeling that it's not easy for you to do that. :(

**Comment by moneytoo** [comment]
My comment was just a general idea as I don't use Kodi. But maybe some "subtitle downloader" app could help.

**Reply by Guguloi** [reply]
OK, I understand. 
Thanks for your time.

**Comment by sigma2017** [comment]
Hello,
Almost everybody use the player over network connection. Me for example I am using solid explorer to be able to connect to my network share and play the files with mxplayer or bsplayer on my AndroidTV but none of them are able to deal with dovi.
I just heard that kodi can play dovi after the latest update but I never liked it because is not lightweight, I don't like it overall.
You player rocks but the solution to load a local subtitle is not fine.
If the .srt will have exactly the same filename as .mkv, it suppose to load it automatically over the network.
It is totally pointless to copy the .srt file from source network share to the TV and after that to start the movie from lan and load the .srt locally from the tv!
This player is too fast, too lightweight, too good to have this annoying subtitle issue. All the players around can do this!

Thank you and congrats for your work!

**Reply by moneytoo** [reply]
You're right, if there's something missing, it's this - #173. I should finish this.

**Reply by moneytoo** [reply]
This feature will be available in the next update. I believe it should work event better than in other players as it can detect not only `.srt` but also file name ending `.en.srt`/`.eng.srt` - based on device language(s).

**Reply by moneytoo** [reply]
It's now in [v0.69](https://github.com/moneytoo/Player/releases/tag/v0.69). Let me know if it works for you.

**Comment by sigma2017** [comment]
Hello,
I have installed v0.69 using the apk and I confirm srt is loaded over the
network.
I think you can update the player version on Google Play as well.
Thanks for your amazing work!

**Reply by sigma2017** [reply]
Just a question regarding ftp. I am using Mixplorer as you have suggested and yes, over sftp(ssh) and samba it works perfectly.
However over ftp the external subtitle is not loaded even the filename is the same for video and subtitle.
I would be nice because ftp (without TLS encryption) is faster on the LAN compared with other protocols like ssh or smb. The idea is to be fast, no need to be secure also on a private LAN.
Some other user requested the same thing and I think it can be very useful!
Thanks again!

**Reply by moneytoo** [reply]
If it doesn't work, then there's not much to do, I'm afraid. Some file explorers just do not allow access to multiple files and other players (like MX Player) also won't be able to load subtitles. I tried multiple file explorers when I was testing this feature and Solid Explorer is still the best one.

**Reply by sigma2017** [reply]
What about sftp? When I stream from my linux box with samba, yes, I have confirmed above, srt is loaded over the network. I tried with Solid explorer, Mixplorer and Es file explorer, all shares the same behavior: srt is loaded just fine with samba connection, with sftp (ssh) it doesn't. Tested with latest 0.76 on my AndroidTV and also on my Android phone.
It would be very very useful if you will fix this also. 
Thank You

**Reply by moneytoo** [reply]
I'm afraid this is a limitation of the combination of some file explorers and protocols. In some cases it just won't work if they do not expose directories in uri path and do not allow access to other files (than those requested). Other video players supporting this fetching of subtitles (like MX Player) also won't work with SFTP. I would recommend using WebDAV (optionally with client certificate).

**Reply by sigma2017** [reply]
Ok, if I  connect sftp I will stick to mx player then. Tested already for some time with solid explorer and Es file explorer using sftp and yes, srt is loaded with mx player. I will use samba with just player, it works since v0.69 when you added this feature.
Thanks for your answer, really appreciated!

**Reply by sigma2017** [reply]
Hello again, you were right, the issue is related to file explorers, not to just player. For example with es-file-explorer-4-2-8-7-1 (latest version) some external .srt are loaded with just player over sftp, others doesn't....couldn't find a match. With older Es-file-explorer-pro-1.1.4.1 all .srt work without any issue in the same scenario. Sorry for the confusion!
